tests/timings_tests.cpp

```diff
@@ -0,0 +1,109 @@
+
+#define ENABLE_PROFILING
+#include "arc_utilities/timing.hpp"
+#include <gtest/gtest.h>
+#include <thread>
+
+
+using namespace arc_utilities;
+
+
+
+TEST(TimerTest, Functioning)
+{
+    Profiler::reset_and_preallocate(10,10);
+    Profiler::startTimer("timer1");
+    double t_elapsed = Profiler::record("timer1");
+    EXPECT_TRUE(t_elapsed > 0) << "Timer recorded negative elapsed time";
+
+    for(size_t i=2; i<100; i++)
+    {
+        Profiler::record("timer1");
+        std::vector<double> times = Profiler::getData("timer1");
+        ASSERT_EQ(i, times.size()) << "Num events and number of times getData called do not match";
+        EXPECT_TRUE(times[i-1] > times[i-2]) << "Time not monotonitcally increasing";
+    }
+
+}
+
+TEST(TimerTest, TimerAccuracy)
+{
+    Profiler::startTimer("sleepy");
+    std::this_thread::sleep_for(std::chrono::milliseconds(50));
+    double t_elapsed = Profiler::record("sleepy");
+    EXPECT_TRUE(t_elapsed < 0.051) << "Sleep for 50ms took too long";
+    EXPECT_TRUE(t_elapsed > 0.0500) << "Recorded time less than sleep time";
+}
+
+TEST(TimerTest, MultiTimer)
+{
+    Profiler::reset_and_preallocate(10,10);
+    Profiler::startTimer("timer1");
+    Profiler::startTimer("timer2");
+    double t2_elapsed = Profiler::record("timer2");
+    double t1_elapsed = Profiler::record("timer1");
+    EXPECT_TRUE(t1_elapsed > t2_elapsed) << "Timer1 started first but not registering longer time";
+    EXPECT_TRUE(t1_elapsed > 0) << "Timer 1 recorded non-positive elapsed time";
```

Comment:
Is there a reason you check that they are positive times as opposed to checking that they are monotonically increasing/accurate/all three tests?

Reply:
In this test I'm checking that creating two timers does indeed create two distinct timers and that the second doesn't just clobber the first.
I'm relying on my other tests for testing monotonicity, accuracy, etc. for a single timer

It's a tradeoff between making complete testing of every combination of cases vs. simple, modifiable code.